### ISSUE TITLE
Avro AVSC compilation fails with version 3.10

### ISSUE THREAD

**Issue by effusion** [quarkusio/quarkus#40621]
### Describe the bug

We use Quarkus (BOM version 3.9.3) in one of our projects. After upgrading to 3.10.0, the build no longer runs and throws an error when generating the Java classes from the *.avsc files. I suspect Smallrye because it's the only lib which currently generates code. The build works in another project where we use the *.avdl files.

### Expected behavior

Compilation works, and the Java classes for the AVSC files are generated.

### Actual behavior

AVSC files placed in the Avro directory can no longer be compiled, and the build crashes with the following exception:
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
        java.lang.IllegalArgumentException: SRCFG00013: No Converter registered for class io.quarkus.maven.dependency.GACT
See the full stacktrace in the build.log file attached.

### How to Reproduce?

1. Create a new Quarkus project with Smallrye messaging.
2. Create the folder "src/main/avro"
3. Add an *.avsc file
4. Execute gradle clean build

### Output of `uname -a` or `ver`

Darwin ultima-thule.localdomain 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:10:42 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk version "21.0.1" 2023-10-17 LTS OpenJDK Runtime Environment Zulu21.30+15-CA (build 21.0.1+12-LTS) OpenJDK 64-Bit Server VM Zulu21.30+15-CA (build 21.0.1+12-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.10.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

------------------------------------------------------------ Gradle 8.7 ------------------------------------------------------------  Build time:   2024-03-22 15:52:46 UTC Revision:     650af14d7653aa949fce5e886e685efc9cf97c10  Kotlin:       1.9.22 Groovy:       3.0.17 Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023 JVM:          21.0.1 (Azul Systems, Inc. 21.0.1+12-LTS) OS:           Mac OS X 14.4.1 aarch64

### Additional information

[build.log](https://github.com/quarkusio/quarkus/files/15308049/build.log)

**Comment by VigneshwaranVenugopal** [comment]
I have the same issue. I am using jdk 17 as the target version in gradle settings

**Comment by mzellho** [comment]
Same here, but:
- not when running `mvn quarkus:dev`
- not when running `mvn clean install`
- only when running the tests from IntelliJ (right click `...src/test/java` -> Run 'Tests in 'java'')

**`uname -a`**
`Linux myhost 6.5.0-28-generic #29-Ubuntu SMP PREEMPT_DYNAMIC Thu Mar 28 23:46:48 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux`

**`java -version`**
```
openjdk version "21.0.3" 2024-04-16 LTS
OpenJDK Runtime Environment Temurin-21.0.3+9 (build 21.0.3+9-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (build 21.0.3+9-LTS, mixed mode, sharing)
```

**Quarkus Version:**
3.10.1

**Comment by ia3andy** [comment]
Just reproduced, with 3.10.1 and kafka-messaging, cc @aloubyansky @cescoffier

**Comment by aloubyansky** [comment]
FYI @radcortez

**Comment by gsmet** [comment]
I suspect Intellij to use something from the previous version and that there is a version mismatch.

**Comment by radcortez** [comment]
Yes, I had similar issues when running in IntelliJ. Usually, I need to build the project first in the command line, nuke the caches, and reimport everything to get everything working correctly again.

Sadly, if everything works fine with the command line builds, then the issue is something in the IDE.

**Comment by effusion** [comment]
I just tried it with the newest version, 3.10.1, as well. It does not matter what tool I use to compile(Intellij, Gradle directly or Gradle on the build server). It always fails with the `No Converter registered` error.

**Comment by gsmet** [comment]
@effusion and versions are consistent? The BOM and the Gradle plugin are both `3.10.1`?

**Comment by aloubyansky** [comment]
I remember seeing a reproducer. There was a discussion about another issue, fixing that the producer would fail with this error. I pinged you there Roberto some time ago. It had a different subject, so not sure i'll be able to find it but it was not an IDE issue.

**Comment by radcortez** [comment]
Unfortunately, I can't remember :(

I was indeed able to reproduce it in the IDE, but not in the command line.

**Comment by cescoffier** [comment]
@effusion  Any update on the reproducer?

**Comment by effusion** [comment]
The Issue resolved itself. It was a wired caching issue on my machine. Although I restarted, cleaned, built, and invalidated the IntelliJ cache, a new git branch on a co-developer's laptop was the resolution. I have no idea what went wrong with my machine.
